aicsimageio/readers/glob_reader.py

```diff
@@ -0,0 +1,384 @@
+        self.chunk_dims = [d.upper() for d in self.chunk_dims]
+        
+        if dim_order is not None:
+            self._dim_order = dim_order
+        else:
+            self._dim_order = "".join(d for d in DEFAULT_DIMENSION_ORDER if d in self._all_files.columns or d in self.chunk_dims)
+
+        self._channel_names = channel_names
+
+        if single_file_shape is None:
+            with self._fs.open(self._path) as open_resource:
+                with TiffFile(open_resource) as tiff:
+                    if tiff.is_shaped:
+                        self._single_file_shape = tuple(
+                            tiff.shaped_metadata[0]["shape"]
+                        )
+                    elif len(tiff.series) == 1:
+                        self._single_file_shape = tiff.series[0].shape
+
+        else:
+            self._single_file_shape = single_file_shape
+
+        if len(single_file_dims) != len(self._single_file_shape):
+            raise exceptions.ConflictingArgumentsError(
+                f"Number of single file dimensions does not match the"
+                f"number of dimensions in a test file. "
+                f"Number of dimensions in file: {len(self._single_file_shape)}, "
+                f"Provided number of dimensions: {len(single_file_dims)}."
+            )
+
+        else:
+            self._single_file_dims = list(single_file_dims)
+
+        self._single_file_sizes = dict(
+            zip(self._single_file_dims, self._single_file_shape)
+        )
+        # Enforce valid image
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        if self._scenes is None:
+            self._scenes = tuple(
+                metadata_utils.generate_ome_image_id(s)
+                for s in range(self._all_files[DimensionNames.Samples].nunique())
+            )
+        return self._scenes
+
+    def _read_delayed(self) -> xr.DataArray:
+
+        scene_files = self._all_files.loc[
+            self._all_files[DimensionNames.Samples] == self.current_scene_index
+        ]
+        scene_files = scene_files.drop(DimensionNames.Samples, axis=1)
+        scene_nunique = scene_files.nunique()
+        group_dims = [
+            x for x in scene_files.columns if x not in self.chunk_dims + ["filename"]
+        ]
+
+        group_sizes = OrderedDict([(d, scene_nunique[d]) for d in group_dims])
+        chunk_sizes = self._get_sizes_for_scene(scene_nunique, group_dims)
+        
+        
+        # Assemble the dask array
+        if len(group_dims)>0: #use groupby to assemble array out of chunks
+            chunks = np.zeros(tuple(group_sizes.values()), dtype="object")
+
+            for i, (idx, val) in enumerate(scene_files.groupby(group_dims)):
+                zarr_im = imread(val.filename.tolist(), aszarr=True)
```

Comment:
You should add: `level=0, chunkmode="page"` to the parameters of `imread`, `chunkmode="page"` generally makes the reading much faster and `level=0` is a safety measure against pyramid tiffs.

Reply:
So i tried this and it seems like `chunkmode="page"` is not implemented for `TiffSequence` which is what gets create when you pass `imread` more than one file. I get an error that says `NotImplementedError: <CHUNKMODE.PAGE: 2> not implemented`. Not sure how essential this is.

Reply:
Ahhhh I see. No worries then. If we run tests of this on real data and the load times feel slow we can revisit these params.